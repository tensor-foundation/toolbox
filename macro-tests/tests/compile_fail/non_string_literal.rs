use solana_program::pubkey::Pubkey;
use tensor_macros::pubkey;

fn main() {
    let invalid = "11111111111111111111111111111111";
    let _pubkey: Pubkey = pubkey!(invalid); // Should fail: not a string literal
}
