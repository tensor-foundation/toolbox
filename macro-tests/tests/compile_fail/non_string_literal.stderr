error: expected string literal
 --> tests/compile_fail/non_string_literal.rs:6:35
  |
6 |     let _pubkey: Pubkey = pubkey!(invalid); // Should fail: not a string literal
  |                                   ^^^^^^^
